articles/per_package_documentation.md

```diff
@@ -40,82 +41,252 @@ The per-package documentation plan is an extension of the general ROS 2 document
+If a package doesn’t have one, the tool can use a default one.
+Python packages will need Sphinx directives explicitly integrated into the source code to produce API documentation (standard for Sphinx API documentation).
+
+Once complete, the tool will output the package documentation build locally.
+Users will be able to see any build errors as well as how their documentation will look once built to `docs.ros.org`.
+
+##### Support for other languages
+
+This section addresses how the system will potentially satisfy requirement 2.4 in the future.
+In general, the goal is to treat additional languages the same as the system treats Python and Doxygen.
+That is, a library in any additional language should be able to utilize its native/standard documentation tool.
+There should then be a bridge between the output of that tool and Sphinx, like how Breathe bridges Doxygen and Sphinx.
+This will most likely not be an out-of-box solution for most languages, but the assumption is that extension will be possible.
+
+Bridging with Sphinx allows the assumption that cross-referencing will be possible.
+It also maintains the standard landing page requirement; the top level entry point will be Sphinx regardless of a package’s language.
+The landing page will provide access to the other generated documentation.
+
+For packages with documentation for multiple languages, the landing pages will provide access to each generator (e.g. “Click here for Doxygen content, Click here for Javadoc content”).
+Supporting multiple documentation engines will require that each generator outputs into its own designated directory (see [URL Structure - /generated](#generated)).
+
+#### Buildfarm process
+
+To opt in to having documentation built on docs.ros.org, maintainers will have to add a `doc` block to their `distribution.yaml` file that indicates the rosdistro branch and a path to the repository.
+Opting-in this way will cause the tool to automatically generate an index landing page for the package.
+
+The buildfarm will invoke the same tool used in the user process, run on ***n*** packages, to generate the documentation for `docs.ros.org`.
+It will require some additional settings, for example the URL where content should generate into.
+
+A webhook between GitHub and the buildfarm will indicate when there is a new commit to a repository (for the branches being tracked for documentation) so the documentation can be automatically regenerated and immediately deployed.
+
+Some files will be auto-generated during this process (changelogs, tag files, etc.).
+The auto-generated content won’t be saved to the upstream repositories, but it will be built to `docs.ros.org`.
+
+There is a possibility of collision occurring between maintainer created files and generated files.
+To prevent this to the best of the system's ability, the system can place auto-generated content in a restricted directory (see [URL Structure - /generated](#generated)).
```

Comment:
```suggestion
To prevent this to the best of the system's ability, the system will place auto-generated content in a restricted directory (see [URL Structure - /generated](#generated)).
```